core/src/test/java/org/infinispan/util/CoreTestBlockHoundIntegration.java

```diff
@@ -56,6 +58,24 @@ public void applyTo(BlockHound.Builder builder) {
             writeJUnitReport(TestResourceTracker.getCurrentTestName(), t, "Uncaught");
          }
       });
+
+      RxJavaPlugins.setErrorHandler(t -> {
+         // Ignore lifecycle exceptions as this can happen when shutting down executors etc.
+         boolean containsIllegalState = false;
+         Throwable cause = t;
+         while (cause != null) {
+            if (cause instanceof IllegalLifecycleStateException) {
```

Comment:
IMO `IllegalLifecycleStateException` should only ever be wrapped in a `CompletionException`, so I'd like to use `CompletableFutures.extractException` instead of the while loop. Or is a RxJava-specific wrapper possible as well?

Reply:
I can try that, but I personally am not fan of assuming how our exceptions are wrapped.

Reply:
Ah actually there was an issue now that I look at the commits on here again https://github.com/infinispan/infinispan/commit/7ba91aaa6b7e2e14b545dece570117e6cc296e9f#diff-3f82a0cc852a40276880a96e45cbc226R64 
I had unwrapped it once and it wasn't sufficient still.

This is an example of one that still failed https://ci.infinispan.org/job/Infinispan/job/PR-8356/4/testReport/junit/org.infinispan.stream/DistributedStreamIteratorReadCommittedTxTest[DIST_SYNC,%20tx=true]/Undelivered/

Reply:
@danberindei WDYT?

Reply:
@wburns I wasn't talking about unwrapping the exception once here, but about taking care that an `IllegalLifecycleStateException` is not wrapped in any other exception elsewhere. IMO unwrapping in a loop here is just hiding the real problem.

That other failure isn't an `IllegalLifecycleStateException`, it's a regular `IllegalStateException`, so I would expect it to be wrapped in a `RemoteException`, and then it's fair to wrap any exception in a `CompletionException`. But then if someone wanted to wrap the exception again, they should first unwrap the `CompletionException`.